horace_core/sqw/@sqw/mask_random_pixels.m

```diff
@@ -47,17 +47,13 @@
     error('HORACE:sqw:invalid_argument', ...
         'Cannot mask every pixel');
 end
-
-if any(npix > arrayfun(@(x) x.pix.num_pixels, win))
+invalid = arrayfun(@(i)(obj(i).pix.num_pixels<npix(i)),1:sz);
```

Comment:
because the previous implementation of the code would have problem if obj array is multidimensional and npix shape ~=1 and is different from the obj shape